felix/dataplane/linux/vxlan_mgr.go

```diff
@@ -105,31 +105,47 @@ func newVXLANManager(
 		blackHoleProto = dpConfig.DeviceRouteProtocol
 	}
 
-	brt := routetable.New(
-		[]string{routetable.InterfaceNone},
-		4,
-		false,
-		dpConfig.NetlinkTimeout,
-		dpConfig.DeviceRouteSourceAddress,
-		blackHoleProto,
-		false,
-		0,
-		opRecorder,
-	)
+	var brt routeTable
+	if !dpConfig.RouteSyncDisabled {
+		log.Info("RouteSyncDisabled is false.")
+		brt = routetable.New(
+			[]string{routetable.InterfaceNone},
+			4,
+			false,
+			dpConfig.NetlinkTimeout,
+			dpConfig.DeviceRouteSourceAddress,
+			blackHoleProto,
+			false,
+			0,
+			opRecorder,
+		)
+	} else {
+		log.Info("RouteSyncDisabled is true, using DummyTable.")
+		brt = &routetable.DummyTable{}
+	}
+
+	noEncapRTConstruct := func(interfaceRegexes []string, ipVersion uint8, vxlan bool, netlinkTimeout time.Duration,
```

Comment:
Maybe move setting of `noEncapRTConstruct` to `if !dpConfig.RouteSyncDisabled` above?